[receiver/mongodbatlas] Add Events API Support (#18237)
open-telemetry/opentelemetry-collector-contrib: receiver/mongodbatlasreceiver/README.md

- `public_key` (required for metrics, logs, or alerts in `poll` mode)
- `private_key` (required for metrics, logs, or alerts in `poll` mode)
- `granularity` (default `PT1M` - See [MongoDB Atlas Documentation](https://docs.atlas.mongodb.com/reference/api/process-measurements/))

> - `storage` configure the component ID of a storage extension. If specified, both alerts `poll` mode and `events` collection will utilize the extension to ensure alerts are not duplicated after a collector restart.

Reviewer: Let's reword this to untangle alerts and events. Maybe:
```suggestion
- `storage` (optional) The component ID of a storage extension which can be used when polling for `alerts` or `events` . The storage extension prevents duplication of data after a collector restart by remembering which data were previously collected.
```

    - `collect_audit_logs` (default false)
    - `include_clusters` (default empty)
    - `exclude_clusters` (default empty)
- `events`
  - `projects`
    - `name` Name of the Project to discover events from
  - `poll_interval` (default `1m`)
    - How often the receiver will poll the Events API for new events.
  - `page_size` (default `100`)
    - This is the number of events that will be processed per request to the MongoDB Atlas API.
  - `max_pages` (default `25`)
    - This will limit how many pages of events the receiver will request from the MongoDB Atlas API for each project.

>   - `types` (default [])

Reviewer: ```suggestion
  - `types` (defaults to all types of events)
```
or something to that effect

>     - This is a list of [event types](https://www.mongodb.com/docs/atlas/reference/api/events-orgs-get-all/#event-type-values) that the receiver will request from the API. If not specified then the receiver will collect all types of events.

Reviewer: ```suggestion
    - This is a list of [event types](https://www.mongodb.com/docs/atlas/reference/api/events-orgs-get-all/#event-type-values) that the receiver will request from the API. If specified, the receiver will collect only the indicated types of events.
```
